server/src/api/bill_exec.py

```diff
@@ -142,3 +142,93 @@ def del_bill(self):
                 'message': error_msg
             }
             return json.dumps(response)
+        
+    @cherrypy.expose    
+    def edit_bill(self):
+        """edits a bill and adds to 
+            the database accordingly
+        
+        JSON Arguments:
+            bill_id -- id of the bill that is being edited
+            title -- user defined name of bill
+            payer -- name of person who fronted bill
+            total -- total bill amount
+            outstanding_payments -- list of outstanding payments
+                person -- name of payer
+                amount -- amount outstanding
```

Comment:
I haven't really thought of a good solution for this hence why it is in the description :) Since the `payment_id` would need to be saved during initialization, there could be some payers with a `payment_id` and others not, which would result in having to make a new list for existing payers or having objects with different properties in the payer list.

Reply:
Ah I see. Sorry - I missed the description :)
Yeah, I can see how it will complicate things.